### DISCUSSION TITLE
hub/app/ios/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#13056] [Docs]
# hub/app/ios/

Transform your iOS device into a powerful object detection tool with the Ultralytics iOS App, powered by YOLO models.

https://docs.ultralytics.com/hub/app/ios/

**Comment by Fireh3x** [comment]
My iPhone X can only load the YOLO 11n model weight which is around 5 MB size. For larger models, the loading takes forever. Is there any size limit for the .pt file to use in iPhone? Thanks.

**Reply by glenn-jocher** [reply]
@Fireh3x the iPhone X's A11 Bionic Neural Engine has limited capacity (0.6 TOPs), which can struggle with larger YOLO models. We recommend using quantized models (FP16 or INT8) to reduce size and improve performance. For more details, refer to the [Ultralytics iOS App documentation](https://docs.ultralytics.com/hub/app/ios/).

**Comment by adripriadana** [comment]
Hello,

I would like to ask about the implementation of YOLO model on iOS platform such as iPhone 11. Is it possible to load the YOLO8n custom model weight trained previously on a PC?

Thank you very much.

Best regards.

**Reply by glenn-jocher** [reply]
@adripriadana yes, you can load your YOLO8n custom model trained on a PC onto the iOS platform using the Ultralytics iOS App. Ensure the model is exported to the CoreML format for compatibility with Apple devices. For more details, refer to the documentation here: https://docs.ultralytics.com/hub/app/ios/.

**Comment by StefanNa3Shape** [comment]
Hi,
Thanks for all the information provided!
I want to test a model on my iPhone 16 through the hub app.
It works fine with all other model that are standard.

Then I trained a model using the hub, made sure it worked in python and exported it in various formats in the hub including CoreML.

I can also load it in the app, but nothing happens.
What is wrong?

The app shows that it is an option to export to Ultralytics iOS, but I can not find that option in the browser hub...

Thank you for the help!

**Comment by cdeil** [comment]
Hi. I'm using the Ultralytics IOS YOLO app. Downloading the models is super slow today, e.g. I clicked "YOLO11x" and it's been loading for over 10 minutes and is now at 70%. I think a few days ago this worked fine and took seconds. Can you check what's wrong? My internet is fast.

**Reply by glenn-jocher** [reply]
Thanks for the report—please keep the details here in this discussion so the Ultralytics team can see and reproduce it; large weights like `YOLO11x` can be sensitive to CDN routing, VPN/iCloud Private Relay, and iOS background download limits (try keeping the app in the foreground, force-quit/reopen, and retry on a different network with VPN/Private Relay off), and if it’s still slow please share your country/region, iOS + app version, Wi‑Fi vs cellular, and whether smaller models download normally (see the broader [Ultralytics HUB app documentation](https://docs.ultralytics.com/platform/) for context on how the app handles models).
